### DISCUSSION TITLE
Lesson 1 challenge help

### DISCUSSION THREAD

**Discussion by phvaz** [Cyfrin/foundry-full-course-cu#913] [Q&A] [answered]
First, I apologize for what could be a stupid question. I'm in lesson 3 by now and went back to complete lesson 1 and 2 challenges, but I'm already stucked, honestly I have no ideia how to proceed or even how to start to solve the challenge. Any tips on what I should do to solve the problem??

Lesson 1 challenge: https://arbiscan.io/address/0x2e99A4CA9c85383ccB54769adB3837437e296479#code

**Comment by theirrationalone** [comment, marked as answer]
You need to call `solveChallenge` function. You can call that function from anywhere either using foundry, remix, or even from etherscan or erbiscan.

Keep this in mind: Calling `solveChallenge` function from foundry, remix. You may need the challenge contract address and its interface to call `solveChallenge` function either from foundry or remix.

Hints: Read Comments Given by patrick just above `solveChallenge` function.

Tip: Connect your metamask with arbiscan or etherscan, After going to `write/read contract tab` either in arbiscan or etherscan and after that you in fact know what the heck is need to do.

;`)

**Reply by phvaz** [reply]
I managed to do the transaction and got my first token! Thanks for your time and patience mate. God bless you. 
![Captura de tela 2023-09-25 102513](https://github.com/Cyfrin/foundry-full-course-f23/assets/116604133/cfa384b4-5387-43a8-b3cb-216e54b7a391)

**Reply by theirrationalone** [reply]
Conquer the Course ❤❤❤

**Reply by leoAshu** [reply]
Hi @phvaz,
I'm glad you could make it work. I am still stuck.
Do we need to have ETH on the mainnet to send this transaction and complete the challenge?
Thanks in advance.

**Comment by Aishwaria8786** [comment]
This question also helped me to solve the first challenge. Thank you.
